sponsors/notifications.py

```diff
@@ -0,0 +1,43 @@
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from django.conf import settings
+
+
+class AppliedSponsorshipNotification:
+    subject_template = None
+    message_template = None
+
+    def get_subject(self, context):
+        return render_to_string(self.subject_template, context).strip()
+
+    def get_message(self, context):
+        return render_to_string(self.message_template, context).strip()
+
+    def get_recipient_list(self, context):
+        raise NotImplementedError
+
+    def notify(self, user, sponsorship):
+        context = {'user': user, 'sponsorship': sponsorship}
+
+        send_mail(
+            subject=self.get_subject(context),
+            message=self.get_message(context),
+            recipient_list=self.get_recipient_list(context),
+            from_email=settings.DEFAULT_FROM_EMAIL,
+        )
+
+
+class AppliedSponsorshipNotificationToPSF(AppliedSponsorshipNotification):
+    subject_template = "sponsors/email/psf_new_application_subject.txt"
+    message_template = "sponsors/email/psf_new_application.txt"
+
+    def get_recipient_list(self, context):
+        return [settings.SPONSORS_TO_EMAIL]
+
+
+class AppliedSponsorshipNotificationToSponsors(AppliedSponsorshipNotification):
+    subject_template = "sponsors/email/sponsor_new_application_subject.txt"
+    message_template = "sponsors/email/sponsor_new_application.txt"
+
+    def get_recipient_list(self, context):
+        return [context["user"].email]
```

Comment:
@ewdurbin I think I missed something here. Just to make sure, this notification should be sent to:

- the logged user;
- all email flagged as primary contact for that sponsor;

Is this correct?

Reply:
Not quite. We want to notify the `User` submitting the `Sponsorship` as well as any `SponsorContact` for which we already have a verified email address, I'll update this shortly.

Reply:
Hm, got it! Both the user and the verified email address would receive the same email, correct?

Also how can I confirm if an email address is a verified one? Should I check if we have users with that email or use some  kind of email verification service?

Reply:
see https://github.com/python/pythondotorg/pull/1679/commits/a5f3d854672a66e3ce36e04a31225819ad312aa2